app/policies/theblog/content_node_policy.rb

```diff
@@ -25,7 +25,7 @@ def unblock?
     end
 
     def account_can_update_record?
-      (user == record.author && account_has_role?(user, [:editor])) ||
+      (user.try(:id) == record.author.try(:id) && account_has_role?(user, [:editor])) ||
```

Comment:
I think since you added an account model you also may add `Account#has_role?` method as well.
Also now we can change relation `author` to the new `Account` model